### PR TITLE
Fixes #1242: reformat files to follow standard

### LINKED ISSUE #1242: Reformat code to follow standard
Ok so once #1241 lands we'll be linting all new code to comply with standard style. Unfortunately automatically reformatting all of the existing code tracked by #1238 seems far from trivial task and still unclear if it's a worthy effort.

Alternative approach would be to just run a lint file by file and address errors programmatically or manually. As far as I can tell `eslint --fix ./path/to/file.js` seems to address majority of errors, but not the comma first issues (see https://github.com/eslint/eslint/issues/6941).

This is also maybe an easy first contribution patch. In case you want to contribute to this issue you should:
1. Remove path for the file you're going to fix from `.eslintignore` file.
2. Run lint and address it's errors.
3. Submit a change via pull request.

Note: This way we can gradually reformat our code.

P.S: No need to reformat stuff in `src/About/Newtab/Demos/` as that code is not maintained in this repo.

### PR DESCRIPTION
**Pull request by magopian**
This PR only deals with the easy fixes:
- move commas to the end of the line
- curly braces placement
- remove unused imports and declaration
- replace `==` and `!=` by `===` and `!==` (when not comparing to `null` or `undefined`)
- various straightforward small fixes

### CODE REVIEW

**Comment by magopian** [comment]
I believe I've been as far as I could with my knowledge of js, flow, standard and the codebase.

I'm not sure how to fix the 7 remaining files, and would welcome any tip or help on those.

Maybe dealing with those problematic ones in another PR would make sense though, this one is already pretty large.

@Gozala any thoughts?

**Comment by Gozala** [comment]
> I believe I've been as far as I could with my knowledge of js, flow, standard and the codebase.

That is amazing! Thank you so much! I'll try to get through this by end of tomorrow!

> I'm not sure how to fix the 7 remaining files, and would welcome any tip or help on those.

I'll definitely try to look into those, but let's get these landed first.

> Maybe dealing with those problematic ones in another PR would make sense though, this one is already pretty large.

Most certainly! I would rather land formatting changes separate from ones that actually alter code, so that if we introduce some regressions it would be easier to track it down and fix.

File: src/Devtools/Log.js [hunk 1]
```diff
@@ -1,76 +1,66 @@
...
 
+import type {Address, DOM} from 'reflex'
 
-import type {Address, Never, DOM, Init, Update, View, AdvancedConfiguration} from "reflex"
-import type {Result} from "../Common/Result"
-import type {URI, ID} from "../Common/Prelude"
-
-
-export type Model <model, action> =
+export type Model <model, action> = // eslint-disable-line no-unused-vars
```
Comment (Gozala) [inline src/Devtools/Log.js]: Question: Why `eslint-disable-line no-unused-vars` is necessary here ? exports aren't considered unused are they ?

File: main.js [hunk 2]
```diff
@@ -1,79 +1,74 @@
...
+        (query === ''
         ? `${arg.substr(2)}`
         : `${query}&${arg.substr(2)}`
         )
       }
     : flag == null  // If there is no flag then skip.
-    ? { flag
-      , query
+    ? { flag,
+       query
```
Comment (magopian) [inline main.js]: I didn't see any indentation "errors" caught by the linter, so I don't think it's even checking it. I'll fix those indentations issues with pleasure ;)

File: src/About/Repl/Repl/Host/Window.js [hunk 3]
```diff
@@ -4,80 +4,73 @@
...
 
+import type {ID, EvaluationResult} from '../Host'
+import type {Never} from 'reflex'
 
-import type {ID, EvaluationResult} from "../Host"
-import type {Never} from "reflex"
-
-
-const DELETE = new String('delete');
+const DELETE = 'delete'
```
Comment (magopian) [inline src/About/Repl/Repl/Host/Window.js]: I'll change it back, but then i'll leave this file in the `standard-flow` ignore list, as it complains:
```
standard-flow: Use JavaScript Standard Style (with Flow types) (https://github.com/gozala/standard-flow)
  /Users/mathieu/mozilla/browserhtml/src/About/Repl/Repl/Host/Window.js:14:16: Do not use String as a constructor.
  /Users/mathieu/mozilla/browserhtml/src/About/Repl/Repl/Host/Window.js:69:57: eval can be harmful.```

File: src/Browser.js [hunk 4]
```diff
@@ -468,289 +428,259 @@ const decodeKeyDown = Keyboard.bindings({
...
+   model,
+   [ ExpandSidebar,
+     ExposeNavigators,
+     SelectPreviousNavigator
     ]
   )
 
 const endSelection = showWebView
 
-const reciveOpenURLNotification = model =>
```
Comment (Gozala) [inline src/Browser.js]: Can you export this instead please.

File: src/Browser/Navigators/Navigator/Assistant.js [hunk 5]
```diff
@@ -252,106 +228,74 @@ export const suggest =
...
-  ( { type: "Suggest"
-    , suggest: suggestion
+  ({ type: 'Suggest',
+     suggest: suggestion
     }
   )
 
-
-
-export const updateSuggestion =
```
Comment (Gozala) [inline src/Browser/Navigators/Navigator/Assistant.js]: Can you please keep this function, I know it's not used but it's going to be soon. And in any case I'd rather prune obsolete code in separate change.

File: src/Browser/Navigators/Navigator/Assistant/Suggestion.js [hunk 6]
```diff
@@ -60,76 +57,43 @@ export const isMatch =
...
-        , match: History.getMatch(query, model.history)
-        , hint: History.getHint(query, model.history)
+          query,
+          match: History.getMatch(query, model.history),
+          hint: History.getHint(query, model.history)
         }
     }
   }
 
-export const update =
```
Comment (Gozala) [inline src/Browser/Navigators/Navigator/Assistant/Suggestion.js]: Can you please keep this function, even if it's not used right now. I would rather remove obsolete code in separate change from formatting one.

File: src/Common/Animation.js [hunk 7]
```diff
@@ -42,152 +40,130 @@ export class Model <model> {
...
+       0,
+       elapsed,
+       duration
       )
-    )
-  , Effects.perform
-    (Task.requestAnimationFrame().map(Tick))
-  ]
-
-const endTransition = <model>
```
Comment (Gozala) [inline src/Common/Animation.js]: Please export this function instead of deleting it.

File: src/Devtools/Log.js (hunk 1 above)
Comment (magopian) [inline src/Devtools/Log.js]: It's the `model` type that is unused, and that's why it's complaining.

**Comment by magopian** [comment]
I believe I've addressed all your comments, please let me know if I missed something or if you'd need anything else, i'd be happy to help!

**Pull request merged**